src/subdomains/supporting/support-issue/services/support-issue.service.ts

```diff
@@ -0,0 +1,99 @@
+import { SupportIssueRepository } from '../repositories/support-issue.repository';
+import { SupportMessageRepository } from '../repositories/support-message.repository';
+import { SupportIssueNotificationService } from './support-issue-notification.service';
+
+@Injectable()
+export class SupportIssueService {
+  constructor(
+    private readonly supportIssueRepo: SupportIssueRepository,
+    private readonly transactionService: TransactionService,
+    private readonly storageService: DocumentStorageService,
+    private readonly userService: UserService,
+    private readonly messageRepo: SupportMessageRepository,
+    private readonly supportIssueNotificationService: SupportIssueNotificationService,
+  ) {}
+
+  async createTransactionIssue(userId: number, transactionId: number, dto: CreateTransactionIssueDto): Promise<void> {
+    const existing = await this.supportIssueRepo.findOneBy({
+      transaction: { id: transactionId },
+      reason: dto.reason,
+    });
+    if (existing) throw new ConflictException('There is already a support issue for this transaction');
+
+    const user = await this.userService.getUser(userId, { userData: true });
+    if (!user.userData.mail) throw new BadRequestException('Mail is missing');
+
+    let entity = this.supportIssueRepo.create({ type: SupportIssueType.TRANSACTION_ISSUE, ...dto });
+
+    entity.transaction = await this.transactionService.getTransactionById(transactionId, { user: { userData: true } });
+    if (!entity.transaction) throw new NotFoundException('Transaction not found');
+    if (!entity.transaction.user || entity.transaction.user.userData.id !== user.userData.id)
+      throw new ForbiddenException('You can only create support issue for your own transaction');
+
+    entity = await this.supportIssueRepo.save(entity);
+
+    await this.createSupportMessage(entity.id, dto, userId);
+  }
+
+  async updateSupportIssue(id: number, dto: UpdateSupportIssueDto): Promise<SupportIssue> {
+    const entity = await this.supportIssueRepo.findOneBy({ id });
+    if (!entity) throw new NotFoundException('Support issue not found');
+
+    Object.assign(entity, dto);
+
+    return this.supportIssueRepo.save(entity);
+  }
+
+  async createSupportMessage(id: number, dto: CreateSupportMessageDto, userId: number): Promise<void> {
+    const existing = await this.messageRepo.findOneBy({
+      message: dto.message,
+      issue: { id },
+    });
+    if (existing) throw new ConflictException('Support message already exists');
+
+    const entity = this.messageRepo.create(dto);
+
+    entity.issue = await this.supportIssueRepo.findOne({
+      where: { id },
+      relations: { transaction: { user: { userData: true } } },
+    });
+    if (!entity.issue) throw new NotFoundException('Support issue not found');
+
+    if (dto.author === CustomerAuthor && entity.issue.user.id !== userId)
+      throw new ForbiddenException('You can only create support messages for your own transaction');
```

Comment:
We need to compare user data IDs (you can create support issues for all your users)